idaes/generic_models/unit_models/tests/test_surrogate_model.py

```diff
@@ -0,0 +1,118 @@
+##############################################################################
+# Institute for the Design of Advanced Energy Systems Process Systems
+# Engineering Framework (IDAES PSE Framework) Copyright (c) 2018-2020, by the
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+Generic template for a surrogate unit model.
+"""
+
+import pytest
+from pyomo.environ import ConcreteModel, Constraint, Var, value, SolverFactory
+from idaes.core import FlowsheetBlock
+from idaes.generic_models.unit_models import SurrogateModel
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.testing import get_default_solver
+
+__author__ = "Jaffer Ghouse"
+
+
+# -----------------------------------------------------------------------------
+# Get default solver for testing
+solver = get_default_solver()
+# -----------------------------------------------------------------------------
+
+m = ConcreteModel()
+m.fs = FlowsheetBlock(default={"dynamic": False})
+
+m.fs.surrogate = SurrogateModel()
+m.fs.surrogate.x_1 = Var(initialize=0.5)
+m.fs.surrogate.x_2 = Var(initialize=0.5)
+m.fs.surrogate.y_1 = Var(initialize=0.5)
+m.fs.surrogate.y_2 = Var(initialize=0.5)
+m.fs.surrogate.c1 = Constraint(
+    expr=m.fs.surrogate.y_1 == m.fs.surrogate.x_1)
+m.fs.surrogate.c2 = Constraint(
+    expr=m.fs.surrogate.y_2 == m.fs.surrogate.x_2**2)
+
+inlet_dict = {"x_1": m.fs.surrogate.x_1,
+              "x_2": m.fs.surrogate.x_2}
+outlet_1_dict = {"y_1": m.fs.surrogate.y_1}
+outlet_2_dict = {"y_2": m.fs.surrogate.y_2}
+
+m.fs.surrogate.add_ports(name="inlet", member_list=inlet_dict)
+m.fs.surrogate.add_ports(name="outlet_1", member_list=outlet_1_dict)
+m.fs.surrogate.add_ports(name="outlet_2", member_list=outlet_2_dict)
+
+
+def my_initialize():
+    # Callback for user provided initialization sequence
+    m.fs.surrogate.c2.deactivate()
+    m.fs.surrogate.x_1.fix(1)
+    m.fs.surrogate.x_2.fix(2)
+
+    opt = SolverFactory("ipopt")
+
+    opt.solve(m)
```

Comment:
Might it be worth recording the solve status of both these solves and  attaching it to the model so that you can asset solver status during the tests (to make sure things worked as expected).